holoviews/core/dimension.py

```diff
@@ -1077,13 +1077,25 @@ def __unicode__(self):
         return unicode(PrettyPrinter.pprint(self))
 
 
-
     def __call__(self, options=None, **kwargs):
+        self.warning('Use opts method instead')
```

Comment:
I'm wary about merging this right straight away. This will overwhelm users with warnings and **a lot** of our docs and examples  are still using ``__call__``. We need to come up with a **very** soft deprecation cycle for this. I'd even say that we should have one release where the only thing nudging users towards using ``.opts`` is consistent usage of ``.opts`` throughout our docs. Anything else is just too disruptive and annoying imo. @jbednar thoughts?